FilePersistence/src/version_control/merge/ChangeGraph.cpp

```diff
@@ -764,7 +764,40 @@ bool ChangeGraph::removeDependenciesInsideNonConflictingAtomicChangeGroups()
 			// This is a cycle which matches all conditions. Remove the dependencies between the elements.
 			removedSomeDependencies = true;
 			for (auto changeToMakeIndependent : changesInCycle)
-				removeAllDependencies(changeToMakeIndependent);
+			{
+				// Nothing will depend on Deletion ( Except Deletion of parentNode )
+				// Deletion will depend on Deletion, MoveOut of children
+				// MoveIn, Insertion of children will depend on Insertion
+				// Insertion will depend on nothing ( Except Insertion of parentNode )
+				if (changeToMakeIndependent->type() != ChangeType::Deletion)
+				{
+					// Remove all Dependencies if change is not deletion
```

Comment:
this comment is inaccurate. If you're really removing all the dependencies as yo say, then why not just call the `removeAllDependencies` method? You're actually filtering some dependencies. Please improve the comment to precisely describe what you're doing.